In [1]:
# using numpy

import numpy as np
x = np.array([3.0,5.0,8.0])
y = np.array([4.0,5.0,9.0])
A = np.vstack([x, np.ones(len(x))]).T   # keep numpy happy as it wants a 2-dimensional X parameter
np.linalg.lstsq(A, y, rcond=None)

(array([1.02631579, 0.52631579]),
 array([0.65789474]),
 2,
 array([10.03106925,  0.6145321 ]))

numpy returns a tuple whose first parameter is an array containing the slope and intercept. The other elements compute various statistics about the quality of the fit. The second parameter is the RSS (Residual sum of squares). It is 0.658 in the above example.

In [2]:
A

array([[3., 1.],
       [5., 1.],
       [8., 1.]])